lib/solveEquation/stepThrough.js

```diff
@@ -251,8 +250,7 @@ function solveConstantEquation(equation, debug, steps=[]) {
   steps = addSimplificationSteps(steps, equation, debug);
   if (steps.length > 0) {
     const lastStep = steps[steps.length - 1];
-    equation = Equation.createEquationFromString(
-      lastStep.newEquation.ascii(), equation.comparator);
+    equation = lastStep.newEquation.clone();
```

Comment:
rather than print the equation back into a string and then reparse it, we can simply clone it